lib/src/testing/test_file.dart

```diff
@@ -0,0 +1,188 @@
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:path/path.dart' as p;
+
+import '../../dart_style.dart';
+
+final _indentPattern = RegExp(r'\(indent (\d+)\)');
+final _fixPattern = RegExp(r'\(fix ([a-x-]+)\)');
+final _unicodeUnescapePattern = RegExp(r'×([0-9a-fA-F]{2,4})');
+final _unicodeEscapePattern = RegExp('[\x0a\x0c\x0d]');
+
+/// Get the absolute local file path to the package's "test" directory.
+Future<String> findTestDirectory() async {
+  var libraryUri = await Isolate.resolvePackageUri(
+      Uri.parse('package:dart_style/src/testing/test_file.dart'));
+  return p.normalize(
+      (p.join(p.dirname(libraryUri!.toFilePath()), '../../../test')));
+}
+
+/// A file containing a series of formatting tests.
+class TestFile {
+  /// Finds all test files in the given directory relative to the package's
+  /// `test/` directory.
+  static Future<List<TestFile>> listDirectory(String name) async {
+    var testDir = await findTestDirectory();
+    var entries = Directory(p.join(testDir, name))
+        .listSync(recursive: true, followLinks: false);
+    entries.sort((a, b) => a.path.compareTo(b.path));
+
+    return [
+      for (var entry in entries)
+        if (entry is File &&
+            (entry.path.endsWith('.stmt') || entry.path.endsWith('.unit')))
+          TestFile._load(entry, p.relative(entry.path, from: testDir))
+    ];
+  }
+
+  /// Reads the test file from [path], which is relative to the package's
+  /// `test/` directory.
+  static Future<TestFile> read(String path) async {
+    var testDir = await findTestDirectory();
+    var file = File(p.join(testDir, path));
+    return TestFile._load(file, p.relative(file.path, from: testDir));
+  }
+
+  /// Reads the test file from [file].
+  factory TestFile._load(File file, String relativePath) {
+    // Explicitly create a File, in case the entry is a Link.
```

Comment:
I don't understand this comment. It looks like this no longer applies after being modified from it's original location.

Reply:
Oops, yes. That comment apparently hasn't been true for a very long time. Removed.